tests/workflow_test.go

```diff
@@ -51,3 +53,31 @@ func (s *e2eSuite) TestWorkflowShow_ReplayableHistory() {
 	err = replayer.ReplayWorkflowHistoryFromJSONFile(nil, historyFile)
 	s.NoError(err)
 }
+
+func (s *e2eSuite) TestWorkflowUpdate() {
+	s.app = app.BuildApp()
+	c := s.ts.Client()
+
+	s.NewWorker(testTq, func(r worker.Registry) {
+		r.RegisterWorkflow(update.Counter)
+	})
+
+	wfr, err := c.ExecuteWorkflow(
+		context.Background(),
+		sdkclient.StartWorkflowOptions{TaskQueue: testTq},
+		update.Counter,
+	)
+	s.NoError(err)
+	signalWorkflow := func() {
+		// send a Signal to stop the workflow
+		err = c.SignalWorkflow(context.Background(), wfr.GetID(), wfr.GetRunID(), update.Done, nil)
+		s.NoError(err)
+	}
+
+	defer signalWorkflow()
+
+	// update
```

Comment:
Worth testing both update accepted and rejected and doing some assertion on the output